server/src/main/java/org/elasticsearch/search/fetch/subphase/InnerHitsPhase.java

```diff
@@ -65,10 +65,6 @@ public void hitsExecute(SearchContext context, SearchHit[] hits) throws IOExcept
                 }
                 innerHits.docIdsToLoad(docIdsToLoad, 0, docIdsToLoad.length);
                 innerHits.setId(hit.getId());
-                innerHits.lookup().source().setSource(context.lookup().source().internalSourceRef());
```

Comment:
We set the root document's _source here, but when processing inner hits we didn't use it and instead reloaded the _source. So this is safe to remove for now.

In a follow-up PR, I will fix this to make sure the _source is loaded only once for the root doc + inner hits.

Reply:
+1, that's something that we spotted sometime ago: https://github.com/elastic/elasticsearch/issues/32818#issuecomment-412878390 and maybe https://github.com/elastic/elasticsearch/issues/56210 but never implemented.